query-engine/connectors/sql-query-connector/src/database/operations/update.rs

```diff
@@ -0,0 +1,179 @@
+    if args.args.is_empty() {
+        return get_single_record(conn, model, &record_filter.filter, &selected_fields, &[], ctx).await;
+    }
+
+    let update = build_update_and_set_query(model, args, Some(&selected_fields), ctx)
+        .so_that(build_update_one_filter(record_filter, ctx));
+
+    let field_names: Vec<_> = selected_fields.db_names().collect();
+    let idents = selected_fields.type_identifiers_with_arities();
+    let meta = column_metadata::create(&field_names, &idents);
+
+    let result_row = conn.query(update.into()).await?.into_iter().next();
+    let record = result_row
+        .map(|row| process_result_row(row, &meta, &selected_fields))
+        .transpose()?
+        .map(|selection| SingleRecord {
+            record: Record::from(selection),
+            field_names: selected_fields.db_names().collect(),
+        });
+
+    Ok(record)
+}
+
+/// Performs an update without an explicit selection set.
+/// This function is called for connectors lacking the `UpdateReturning` capability.
+/// As we don't have a selection set to work with, this function always returns a record with the primary identifier of the model (provided that a record was found).
+/// However, since we can't get the updated values back from the update operation, we need to read the primary identifier _before_ the update and then update the ids in-memory if they were updated.
+pub(crate) async fn update_one_without_selection(
+    conn: &dyn Queryable,
+    model: &Model,
+    record_filter: RecordFilter,
+    args: WriteArgs,
+    ctx: &Context<'_>,
+) -> crate::Result<Option<SingleRecord>> {
+    // If there's nothing to update, just return the ids.
+    // If the parent operation did not pass any ids, then perform a read so that the following operations can be resolved.
+    if args.args.is_empty() {
+        let ids: Vec<SelectionResult> = conn.filter_selectors(model, record_filter.clone(), ctx).await?;
+
+        let record = ids.into_iter().next().map(|id| SingleRecord {
+            record: Record::from(id),
+            field_names: model.primary_identifier().db_names().collect_vec(),
+        });
+
+        return Ok(record);
+    }
+
+    // Pick the primary identifiers args from the WriteArgs if there are any.
+    let id_args = pick_args(&model.primary_identifier().into(), &args);
+    // Perform the update and return the ids on which we've applied the update.
+    // Note: We are _not_ getting back the ids from the update. Either we got some ids passed from the parent operation or we perform a read _before_ doing the update.
+    let (_, ids) = update_many_from_ids_and_filter(conn, model, record_filter, args, ctx).await?;
+    // Since we could not get the ids back from the update, we need to apply in-memory transformation to the ids in case they were part of the update.
+    // This is critical to ensure the following operations can operate on the updated ids.
+    let merged_ids = merge_write_args(ids, id_args);
```

Comment:
Again, highly appreciating the comments, which weren't there in the code in the `write` module before the refactoring.